crates/ide_db/src/source_change.rs

```diff
@@ -85,6 +85,7 @@ impl From<FxHashMap<FileId, TextEdit>> for SourceChange {
 pub enum FileSystemEdit {
     CreateFile { dst: AnchoredPathBuf, initial_contents: String },
     MoveFile { src: FileId, dst: AnchoredPathBuf },
+    MoveDir { src: AnchoredPathBuf, src_id: Option<FileId>, dst: AnchoredPathBuf },
```

Comment:
I was thinking from `MoveDir`'s perspective, in the future, maybe you only want to move dir from `src` to `dst` and the operation itself isn't associated to a specific file. It's totally enough to keep this as plain `FileId` for now and change later.

Reply:
Ah okay, its probably better to keep it as `FileId` until needed then. Otherwise it might be confusing to someone reading the code (or they may just straight up remove it themself)